src/components/Common/CaseStudyCards/CaseStudyBlocks.js

```diff
@@ -0,0 +1,91 @@
+import React, { Fragment } from 'react'
+import ReactMarkdown from 'react-markdown'
+import styled from 'styled-components'
+import breakpoint from 'styled-components-breakpoint'
+
+import { Col } from '../../../components/grid'
+import { SectionTitle, BodyPrimary } from '../../../components/Typography'
+import Image from '../../../components/Common/Image'
+import VideoSection from '../../../components/Common/VideoSection'
+
+const StyledColumnImage = styled(Col)`
+  padding-bottom: ${({ theme }) => theme.space[3]};
+
+  ${breakpoint('smallTablet')`
+    padding-bottom: 0;
+  `}
+`
+
+const renderImage = data => data && <Image image={data} />
+
+const renderText = (text, colorReverse, disallowed = []) =>
+  text && (
+    <ReactMarkdown
+      disallowedTypes={disallowed}
+      renderers={{
+        // eslint-disable-next-line
+        heading: props => <SectionTitle reverse={colorReverse} {...props} />,
+        // eslint-disable-next-line
+        paragraph: props => <BodyPrimary reverse={colorReverse} {...props} />
+      }}
+      source={text}
+    />
+  )
+
+const TextColumnsBlock = ({ data: { text }, colorReverse = false }) => (
+  <Fragment>
+    <Col width={[1, 1, 1, 1, 1 / 2]}>
+      {renderText(text, colorReverse, ['paragraph'])}
+    </Col>
+    <Col width={[1, 1, 1, 1, 1 / 2]}>
+      {renderText(text, colorReverse, ['heading'])}
+    </Col>
+  </Fragment>
+)
+
+const TextAndImageBlock = ({ data: { text, image }, colorReverse = false }) => (
+  <Fragment>
+    <Col width={[1, 1, 1, 1, 1 / 2]}>{renderText(text, colorReverse)}</Col>
+    <Col width={[1, 1, 1, 1, 1 / 2]}>{renderImage(image)}</Col>
+  </Fragment>
+)
+
+const FullWidthBlock = ({ data: { text, image } }) => (
+  <Col width={[1]}>
+    {renderText(text)}
+    {renderImage(image)}
+  </Col>
+)
+
+const VideoBlock = ({ data: { text } }) => <VideoSection src={text} />
+
+const ImagesBlock = ({ data }) => {
+  const l = data.length
+  const lWithGap = l + 1
+  const colWidth = [
+    2 / lWithGap,
+    2 / lWithGap,
+    1 / lWithGap,
+    1 / lWithGap,
+    1 / l,
+    1 / l
+  ]
```

Comment:
This is the kind of stuff that was I had in mind when advising not to make a one size fits all solution. It certainly works but I can't see the advantage of this over something like 

`{images && images.length && images.map((image) => <Col width={[1, 1, 1, 1, 3 / 12]}><Image {...image}/></Col>)}`

which is where we're using this for

Reply:
Yeah; I will remove the possibility of customising this component. You are right: It might be added logic that we don't need at the moment.